src/mame/namco/namcos10.cpp

```diff
@@ -705,26 +887,50 @@ class namcos10_memp3_state : public namcos10_state
 
 	void init_g13jnc();
 
+protected:
+	virtual void machine_start() override;
+	virtual void machine_reset() override;
+
 private:
-	void nand_copy(uint8_t *nand_base, uint16_t *dst, uint32_t address, int len);
-	void memp3_driver_init();
+	void namcos10_memp3_map_inner(address_map &map);
+	void namcos10_memp3_map(address_map &map);
+	void mcu_map(address_map &map);
 
-	optional_device_array<nand_device, 16> m_nand;
-	optional_device<tmp95c061_device> m_memp3_mcu;
+	void firmware_write_w(uint16_t data);
+
+	void ram_bank_w(uint16_t data);
+
+	uint16_t unk_status1_r();
+	uint16_t unk_status2_r();
+
+	uint16_t ram_r(offs_t offset);
+	void ram_w(offs_t offset, uint16_t data);
+
+	uint16_t analog_io_r(offs_t offset);
+
+	required_device<tmp95c061_device> m_memp3_mcu;
+	required_shared_ptr<uint16_t> m_mcu_ram;
+	optional_ioport_array<4> m_p3_analog;
+
+	uint16_t m_mcu_bank;
 };
 
 ///////////////////////////////////////////////////////////////////////////////////////////////
 
+void namcos10_state::psx_remap_io(s32 param)
+{
+	if (!m_psx_remapper)
+		return;
```

Comment:
You can do the check before calling `synchronize` (or possibly even before subscribing to changes) and avoid unnecessary trips through the scheduler.